triton-constraint-circuit/src/lib.rs

```diff
@@ -872,6 +870,14 @@ impl<II: InputIndicator> ConstraintCircuitMonad<II> {
             .count()
     }
 
+    /// Returns a mapping from node ID to the node's degree.
+    fn all_degrees(multicircuit: &[ConstraintCircuit<II>]) -> HashMap<usize, isize> {
+        Self::all_nodes_in_multicircuit(multicircuit)
+            .into_iter()
+            .map(|node| (node.id, node.degree()))
+            .collect()
+    }
+
```

Comment:
Inlining this used-once method keeps things clearer, in my opinion: not two different names for the same thing (`node_degrees` vs `all_degrees`), more type inference, fewer comments needed.

```suggestion
```